R/zchunk_L115.nh3_an_R_S_T_Y.R

```diff
@@ -31,47 +29,112 @@ module_emissions_L115.nh3_an_R_S_T_Y_DISABLED <- function(command, ...) {
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Compute emissions using EPA emissions factors and FAO animal production
+
+    # Rename column and add gas name
+    L107.an_Prod_Mt_R_C_Sys_Fd_Y %>%
+      rename(production = value) %>%
+      mutate(Non.CO2 = "NH3_AGR") ->
+      L115.nh3_tg_R_an_C_Sys_Fd_Yh
+
+    # Match emissions factors by year and compute unscaled emissions
+    L115.nh3_tg_R_an_C_Sys_Fd_Yh %>%
+      left_join(L105.nh3_tgmt_USA_an_Yh, by = "year") %>%
+      rename(emfact = value) %>%
+      mutate(epa_emissions = production * emfact) %>%
+      na.omit() ->
+      L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt
+
+    # Match by sector, technology and fuel
+    left_join(tidyr::unite(L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt, col = "commodity_system_feed", c(GCAM_commodity, system, feed), sep = "~"),
+              tidyr::unite(GCAM_sector_tech, col = "sector_tech_fuel", c(sector, fuel, technology), sep = "~"),
+              by = c("commodity_system_feed"="sector_tech_fuel")) %>%
+      tidyr::separate(commodity_system_feed, c("GCAM_commodity", "system", "feed"), sep = "~") %>%
+      select(GCAM_region_ID, GCAM_commodity, system, feed, year, production, Non.CO2, emfact, epa_emissions, EDGAR_agg_sector) ->
+      L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt
+
+    # Aggregate by sector and region
+    L115.nh3_tg_R_an_C_Sys_Fd_Yh.mlt %>%
+      group_by(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year) %>%
+      summarize(EPA_emissions = sum(epa_emissions)) %>%
+      arrange(year) ->
+      L115.nh3_tg_R_an_C_Yh.mlt
+
+    # Compute EDGAR emissions by region and sector
+    L115.EDGAR <- EDGAR_NH3
+
+    # Renaming L115.EDGAR variable names - Note: may remove if names for L115.EDGAR change
+    names(L115.EDGAR)[1] <- "IPCC_Annex"
+    names(L115.EDGAR)[2] <- "World_Region"
+
+    # Add gas name and match by emissions factors
+    L115.EDGAR %>%
+      mutate(Non.CO2 = "NH3_AGR") %>%
+      left_join(EDGAR_sector, by = "IPCC") %>%
+      rename(EDGAR_agg_sector = agg_sector) ->
+      L115.EDGAR
+
+    # Convert from EDGAR iso to GCAM_region_ID
+    L115.EDGAR %>%
+      mutate( iso = tolower( ISO_A3 ), ISO_A3 = NULL ) %>%
+      change_iso_code('rou', 'rom') %>%
+      left_join(iso_GCAM_regID, by = "iso") ->
```

Comment:
Should be `left_join_error_no_match`? In general (e.g. lines 56, 80, 106, 114) if you can use a more restrictive join, please do so.